support/Tests/GraphQLSupportTests/IntegrationTests.swift

```diff
@@ -56,6 +56,24 @@ class IntegrationTests: XCTestCase {
         XCTAssertEqual(queryString, "mutation{set_integer(input:{key:\"answer\",value:42,negate:true})}")
     }
 
+    func testInputObjectExplictNilConstructor() {
+        let query = Generated.buildMutation { $0
+            .setInteger(input: Generated.SetIntegerInput(key: "answer", value: 42, negate: .some(nil)))
```

Comment:
`Optional.none` is an optional with nil, `Optional.some(nil)` is saying that the optional has a value, but when you unwrap it is nil. That would be making it a double optional like your description `Optional(Optional(value))`

```swift
var a: String? = .none
print(a)

var b: String?? = ""
print(b)

// output
nil
Optional(Optional(""))
```